spacy/kb_base.pyx

```diff
@@ -0,0 +1,168 @@
+    kb (BaseKnowledgeBase): Knowledge base to query.
+    mention (Iterable[Union[Span, str]]): Entity mentions for which to identify candidates.
+    RETURNS (Iterable[Iterator[Candidate]]): Identified candidates.
+    """
+    return kb.get_candidates_batch(mentions)
+
+
+def get_candidates(kb: BaseKnowledgeBase, mention: Union[Span, str]) -> Iterator[Candidate]:
+    """
+    Return candidate entities for a given mention and fetching appropriate entries from the index.
+    kb (BaseKnowledgeBase): Knowledge base to query.
+    mention (Union[Span, str]): Entity mention for which to identify candidates.
+    RETURNS (Iterator[Candidate]): Identified candidates.
+    """
+    return kb.get_candidates(mention)
+
+
+cdef class BaseKnowledgeBase:
+    """A `BaseKnowledgeBase` instance stores unique identifiers for entities and their textual aliases,
+    to support entity linking of named entities to real-world concepts.
+    This is an abstract class and requires its operations to be implemented.
+
+    DOCS: https://spacy.io/api/kb
+    """
+
+    def __init__(self, vocab: Vocab, entity_vector_length: int):
+        """Create a KnowledgeBase."""
+        self.vocab = vocab
+        self._entity_vector_length = entity_vector_length
+        self.mem = Pool()
+
+    property entity_vector_length:
+        def __get__(self):
+            return self._entity_vector_length
+        def __set__(self, int value):
+            self._entity_vector_length = value
+
+    def get_candidates_batch(self, mentions: Union[Iterable[Span], Iterable[str]]) -> Iterable[Iterator[Candidate]]:
+        """
+        Return candidate entities for specified texts. Each candidate defines the entity, the original alias,
+        and the prior probability of that alias resolving to that entity.
+        If the no candidate is found for a given text, an empty list is returned.
+        mentions (Union[Iterable[Span], Iterable[str]]): Mentions for which to get candidates.
+        RETURNS (Iterable[Iterator[Candidate]]): Identified candidates.
+        """
+        return [self.get_candidates(span) for span in mentions]
+
+    def get_candidates(self, mention: Union[Span, str]) -> Iterator[Candidate]:
+        """
+        Return candidate entities for specified text. Each candidate defines the entity, the original alias,
+        and the prior probability of that alias resolving to that entity.
+        If the no candidate is found for a given text, an empty list is returned.
+        mention (Union[Span, str]): Mention for which to get candidates.
+        RETURNS (Iterator[Candidate]): Identified candidates.
+        """
+        raise NotImplementedError
+
+    def get_vectors(self, entities: Iterable[str]) -> Iterable[Iterable[float]]:
+        """
+        Return vectors for entities.
+        entity (str): Entity name/ID.
+        RETURNS (Iterable[List[float]]): Vectors for specified entities.
```

Comment:
The Docstring and type annotation disagree about the return type. The Docstring is more specific and not wrong, but I guess you want the type annotation one here for future flexibility.